### PR TITLE
Refactor btsieve

### PR DESCRIPTION
**Pull request by tcharding**
For the first out-of-sprint-scope day I chose to spend some time learning the `btsieve` codebase.  This PR is a result of that time.  Refactoring is limited to the config file / settings code including command line options.

Stripped down after comments from @thomaseizinger offline.

Also fixes: #1259 (tested on Linux, needs testing on OSX please @D4nte @da-kami or @bonomat)

### CODE REVIEW

**Comment by D4nte** [comment]
> Of note, this adds a new crate in `internal/`.

What is the destiny of this new crate please? I am not sure we want something to just "stay there".

File: internal/comit_directories/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+use directories::ProjectDirs;
+use std::path::PathBuf;
+
+// Linux: /home/<user>/.config/<name>/
```
Comment (D4nte) [inline internal/comit_directories/src/lib.rs]: Any chance you'd be keen to do #1259 at the same time?

**Comment by tcharding** [comment]
FTR @thomaseizinger and I discussed this PR offline.  Posting here for reference.  It does not add much value to have a separate crate to wrap `directories` and it adds some crate management overhead.  Let's just duplicate the code in `cnd` and `btsieve`.

**Comment by tcharding** [comment]
> > Of note, this adds a new crate in `internal/`.
> 
> What is the destiny of this new crate please? I am not sure we want something to just "stay there".

No longer adding this crate, thanks.

File: internal/comit_directories/src/lib.rs (hunk 1 above)
Comment (tcharding) [reply internal/comit_directories/src/lib.rs]: Just got made aware of this comment now, will do.

File: btsieve/src/load_settings.rs [hunk 2]
```diff
@@ -46,3 +35,10 @@ fn parse_config_file(file: PathBuf) -> Result<Settings, ConfigError> {
         )))
     }
 }
+
+// Linux: /home/<user>/.config/comit/
+// Windows: C:\Users\<user>\AppData\Roaming\comit\config\
+// OSX: /Users/<user>/Library/Preferences/comit/
+fn config_dir() -> Option<PathBuf> {
+    ProjectDirs::from("", "", "comit").map(|proj_dirs| proj_dirs.config_dir().to_path_buf())
```
Comment (D4nte) [inline btsieve/src/load_settings.rs]: ❤️

**Pull request merged**